makefiles/targets/_common/darwin_head.mk

```diff
@@ -60,25 +55,46 @@ else
 	__invocation = $(1)
 endif
 
-TARGET_CC ?= $(call __invocation,$(_THEOS_TARGET_CC))
-TARGET_CXX ?= $(call __invocation,$(_THEOS_TARGET_CXX))
-TARGET_LD ?= $(call __invocation,$(_THEOS_TARGET_CXX))
-TARGET_SWIFT ?= $(call __invocation,$(_THEOS_TARGET_SWIFT))
-TARGET_LIPO ?= $(call __invocation,lipo)
-TARGET_STRIP ?= $(call __invocation,strip)
-TARGET_CODESIGN_ALLOCATE ?= $(call __invocation,codesign_allocate)
-TARGET_LIBTOOL ?= $(call __invocation,libtool)
-TARGET_XCODEBUILD ?= $(call __invocation,xcodebuild)
-TARGET_XCPRETTY ?= $(call __invocation,xcpretty)
+# give precedence to Swift toolchains located at SWIFTBINPATH
+ifeq ($(call __exists,$(SWIFTBINPATH)),$(_THEOS_TRUE))
+	__invocation_swift = $(SWIFTBINPATH)/$(1)
+else
+	__invocation_swift = $(call __invocation,$(1))
+endif
+
+ifeq ($(_THEOS_IS_MAKE_GT_4_0),)
+# Make 3 has a bug where __simplify ends up truncating text sometimes
+# (root cause presently unknown). Only use __simplify for targets on
+# Make >= 4
+	__target_simplify = $(2)
+else
+	__target_simplify = $(call __simplify,$(1),$(2))
+endif
+
+TARGET_CC ?= $(call __target_simplify,TARGET_CC,$(call __invocation,$(_THEOS_TARGET_CC)))
+TARGET_CXX ?= $(call __target_simplify,TARGET_CXX,$(call __invocation,$(_THEOS_TARGET_CXX)))
+TARGET_LD ?= $(call __target_simplify,TARGET_LD,$(call __invocation,$(_THEOS_TARGET_CXX)))
+TARGET_LIPO ?= $(call __target_simplify,TARGET_LIPO,$(call __invocation,lipo))
+TARGET_STRIP ?= $(call __target_simplify,TARGET_STRIP,$(call __invocation,strip))
+TARGET_CODESIGN_ALLOCATE ?= $(call __target_simplify,TARGET_CODESIGN_ALLOCATE,$(call __invocation,codesign_allocate))
+TARGET_LIBTOOL ?= $(call __target_simplify,TARGET_LIBTOOL,$(call __invocation,libtool))
+TARGET_XCODEBUILD ?= $(call __target_simplify,TARGET_XCODEBUILD,$(call __invocation,xcodebuild))
+TARGET_XCPRETTY ?= $(call __target_simplify,TARGET_XCPRETTY,$(call __invocation,xcpretty))
+TARGET_SWIFTC ?= $(call __target_simplify,TARGET_SWIFTC,$(call __invocation_swift,$(_THEOS_TARGET_SWIFTC)))
+
+TARGET_SWIFT_SUPPORT_BIN ?= $(THEOS_VENDOR_SWIFT_SUPPORT_PATH)/.theos_build/release
+TARGET_ORION_BIN ?= $(THEOS_VENDOR_ORION_PATH)/.theos_build/release
 
 TARGET_STRIP_FLAGS ?= -x
 
 ifeq ($(TARGET_DSYMUTIL),)
-ifeq ($(call __executable,$(call __invocation,dsymutil)),$(_THEOS_TRUE))
-	TARGET_DSYMUTIL = $(call __invocation,dsymutil)
-else ifeq ($(call __executable,$(call __invocation,llvm-dsymutil)),$(_THEOS_TRUE))
-	TARGET_DSYMUTIL = $(call __invocation,llvm-dsymutil)
-endif
+	TARGET_DSYMUTIL := $(call __invocation,dsymutil)
```

Comment:
Since `dysmutil` is vended by both Xcode and the Linux/iOS toolchains, would it be worth assigning `TARGET_DSYMUTIL` along with the other TARGETs above and reassigning only if it's blank (for whatever reason)?

Reply:
This is one of those cases where the answer is "that sounds reasonable" but due to the non-standard `llvm-dsymutil` checking going on here I suspect that there's a reason for structuring the code this way; and imo when it comes to GNU Make if it ain't broke don't fix it